### PR TITLE
add odla_popart code

### PR DESCRIPTION
**Pull request by yanwei-gr**
Add odla_popart code to support IPU.

### CODE REVIEW

**Comment by weimingzha0** [comment]
I need to upgrade the buildbot docker image to include the SDK.

File: ODLA/platforms/odla_popart/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+target_compile_options(custom_ops PRIVATE ${CUSTOM_OP_CXX_FLAGS})
+
+#set(POPLAR_ROOT ${POPLAR_SDK_ENABLED})
+#set(POPART_ROOT ${POPART_ROOT})
+set(ODLA_POPART_PATH ${CMAKE_CURRENT_SOURCE_DIR}/ODLA/platforms/odla_popart)
+add_odla_library(odla_popart SHARED odla_compute.cc odla_ops.cc common.cc) 
+ #find_library(popart NAMES popart PATHS ${POPART_ROOT} PATH_SUFFIXES lib NO_DEFAULT_PATH)
+ #find_library(popart NAMES poplar PATHS ${POPLAR_ROOT} PATH_SUFFIXES lib NO_DEFAULT_PATH)
+ #find_library(popart NAMES popops PATHS ${POPLAR_ROOT} PATH_SUFFIXES lib NO_DEFAULT_PATH)
+find_library(popart NAMES custom_ops PATHS ${ODLA_POPART_PATH}/custom_ops NO_DEFAULT_PATH)
```
Comment (weimingzha0) [inline ODLA/platforms/odla_popart/CMakeLists.txt:34]: Can we add a check? If "find_library" fails, the build will skip the build of odla_popart? (Similar to what we did for TensorRT:
https://github.com/alibaba/heterogeneity-aware-lowering-and-optimization/blob/master/ODLA/platforms/CMakeLists.txt)

So users without Poplar SDK won't hit the build error.

**Comment by youbeny** [comment]
custom op related external head files will be pre-installed in docker images

File: ODLA/platforms/odla_popart/CMakeLists.txt (hunk 1 above)
Comment (yanwei-gr) [reply ODLA/platforms/odla_popart/CMakeLists.txt:34]: Done

**Review by youbeny — APPROVED** [review]
looks good to me

**Review by weimingzha0 — APPROVED** [review]

**Pull request merged**